impl/client/src/main/javascript/web/util/URLEncoder.js

```diff
@@ -57,7 +57,15 @@ define("common-ui/util/URLEncoder", [], function() {
 
   var O_proto = Object.prototype;
 
-  Encoder.encode = function(str, args, queryObj) {
+  /**
+   * We want to preserve the old behavior of this function here, as we have overloaded it to disable double-encoding.
+   * We do not wish to double-encode slashes in all cases, e.g. URL parameters that are paths.
+   */
+  Encoder.encode = function(str, args, queryObj){
```

Comment:
In JavaScript, we cannot create overloads in this way. The second overload function, below, simply overwrites the first one here set on the `Encoder.encode` property. In JavaScript, functions are values.

The alternative pattern is then to have all needed parameters in a function and to explicitly check if these were specified by comparing them to `undefined`, or, in the case of a boolean property, more leniently checking if it is _nully,_ `== null` (which matches both `null` and `undefined`).

The way this is proposed, the default value of `doubleEncodeSlashes` is, in practice, inadvertently changed to `false`.